regression-test/suites/load_p0/insert/test_insert.groovy

```diff
@@ -48,4 +48,38 @@ suite("test_insert") {
     """
 
     qt_sql1 "select * from ${insert_tbl} order by 1, 2, 3, 4"
+
+    def insert_tbl_dft = "test_insert_dft_tbl"
+    sql """ DROP TABLE IF EXISTS ${insert_tbl_dft}"""
+    
+    // `k7` should be float type, and bug exists now, https://github.com/apache/doris/pull/20867
+    // `k9` should be char(16), and bug exists now as error msg raised:"can not cast from origin type TINYINT to target type=CHAR(16)" when doing insert
+    // "`k13` datetime default CURRENT_TIMESTAMP" might have cast error in strict mode when doing insert:
+    // [INTERNAL_ERROR]Invalid value in strict mode for function CAST, source column String, from type String to type DateTimeV2
```

Comment:
I added log in function_cast and the log shows that the cast function get string `CURRENT_TIMESTAMP`:
```
mysql [test]>insert into dft values();
ERROR 1105 (HY000): errCode = 2, detailMessage = (10.16.10.8)[CANCELLED][INTERNAL_ERROR]Invalid value CURRENT_TIMESTAMP in strict mode for function CAST, source column String, from type String to type DateTimeV2
```

Reply:
You can modify the code and try:
```
                                return Status::InternalError(
                                        "Invalid value {} in strict mode for function {}, source "
                                        "column {}, from "
                                        "type "
                                        "{} to type {}",
                                        std::string((char*)&(*chars)[current_offset], string_size), Name::name, col_from->get_name(), FromDataType().get_name(),
                                        ToDataType().get_name());
```

Reply:
Seems the plan is wrong:
```
mysql [test]>explain verbose insert into dft values();
+---------------------------------------------------------------------------------------+
| Explain String                                                                        |
+---------------------------------------------------------------------------------------+
| PLAN FRAGMENT 0                                                                       |
|   OUTPUT EXPRS:                                                                       |
|      CAST('CURRENT_TIMESTAMP' AS DATETIMEV2(0))                                       |
|   PARTITION: UNPARTITIONED                                                            |
|                                                                                       |
|   OLAP TABLE SINK                                                                     |
|     TUPLE ID: 0                                                                       |
|     RANDOM                                                                            |
|                                                                                       |
|   0:VUNION                                                                            |
|      constant exprs:                                                                  |
|          CAST('CURRENT_TIMESTAMP' AS DATETIMEV2(0))                                   |
|      tuple ids: 1                                                                     |
|                                                                                       |
| Tuples:                                                                               |
| TupleDescriptor{id=0, tbl=null, byteSize=16}                                          |
|   SlotDescriptor{id=0, col=k13, colUniqueId=0, type=datetimev2(0), nullable=true}     |
|                                                                                       |
| TupleDescriptor{id=1, tbl=null, byteSize=16}                                          |
|   SlotDescriptor{id=1, col=null, colUniqueId=null, type=datetimev2(0), nullable=true} |
+---------------------------------------------------------------------------------------+
20 rows in set (0.01 sec)
```

Reply:
I got it, the default value here for `CURRENT_TIMESTAMP` is the string itself, which is confirmed to be a bug. https://github.com/apache/doris/blob/bf197ee8d2d9e6ccf88184663e4614068bc3edde/fe/fe-core/src/main/java/org/apache/doris/analysis/NativeInsertStmt.java#L661C52-L675